### PR TITLE
cicd: parallelize full smoke test

### PR DESCRIPTION
**Pull request by midleman**
### Intent

The goal here is to speed up our Full Test Suite by running all tests (integration, unit, and e2e) in parallel, instead of serially.

### Approach

Each test now has its own runner, so they can all run all at once instead of waiting on each other. For now, the build and compile steps are still happening in each test job, but in the future, we could optimize this further by moving those steps to an upstream job and sharing the built artifacts with the test jobs that follow.

#### Previously ~ 54min
Screenshot 2024-10-10 at 8 23 43 AM

#### After changes ~ 27min
Screenshot 2024-10-10 at 8 22 37 AM

### QA Notes
* Confirmed PR tests pass
* Confirmed full test suite tests pass
* I did not test slack notifications, but I made NO changes to the logic/behavior.
* Kept running into failures/flakes from the interpreter dropdown test, so skipped it for now and will look into after this is merged.

### CODE REVIEW

File: .github/actions/install-r/action.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        echo "Installing R version $R_VERSION using Rig..."
+        curl -Ls https://github.com/r-lib/rig/releases/download/latest/rig-linux-"$(arch)"-latest.tar.gz | sudo tar xz -C /usr/local
+        rig add "$R_VERSION"
+
+    - name: Check R Installation
+      shell: bash
+      run: |
+        R --version
+
+    - name: Conditionally Install R Packages
```
Comment (testlabauto) [inline .github/actions/install-r/action.yml:25]: Out of curiosity, why is this called out as conditional?

Comment (midleman) [reply .github/actions/install-r/action.yml:25]: oh! i can remove that. i thought the pkgs weren't always needed in some cases, but i think they are. let me update that.

**Review by testlabauto — APPROVED** [review]
LGTM!

File: .github/workflows/positron-full-test.yml [hunk 2]
```diff
@@ -132,68 +42,104 @@ jobs:
...
+        with:
+          node-version-file: .nvmrc
+
+      - name: Cache node_modules, build, extensions, and remote
+        uses: ./.github/actions/cache-multi-paths
+
+      - name: Setup Build and Compile
+        uses: ./.github/actions/setup-build-env
+
+      - name: Install Positron License
```
Comment (jonvanausdeln) [inline .github/workflows/positron-full-test.yml:64]: Do integration tests need the license?

Comment (testlabauto) [reply .github/workflows/positron-full-test.yml:64]: Yeah, the remote ones do.

**Review by jonvanausdeln — APPROVED** [review]
LGTM! Just one comment/question, but :shipit:

File: .github/workflows/positron-full-test.yml (hunk 2 above)
Comment (midleman) [reply .github/workflows/positron-full-test.yml:64]: Yep, what @testlabauto said.

**Pull request merged**